docs: describe how to manage embedded plugin registry for VS Code editor (#2503)
codenvy/che-docs: modules/administration-guide/partials/proc_importing-extensions-from-the-open-vsx-registry.adoc

[id="importing-extensions-from-the-open-vsx-registry"]
= Importing extensions from the Open VSX Registry

> This section describes how to add VS Code extensions from link:https://open-vsx.org/[Open VSX Registry] to {prod-short}'s embedded Open VSX Registry.

Reviewer: ```suggestion
This section describes how to add Visual Studio Code extensions from link:https://open-vsx.org/[Open VSX Registry] to {prod-short}'s embedded Open VSX Registry.
```

.Prerequisites

. The extension should be published on link:https://open-vsx.org/[Open VSX Registry]

> . An information about an ID of the extension. The ID contains the extension's pubisher and name (e.g. `redhat.vscode-xml`).

Reviewer: ```suggestion
. The extension's ID, which is composed of publisher name and extension name, must be known (e.g. `redhat.vscode-xml`).
```

.Procedure

> . Modify `openvsx-sync.json` file in the link:https://github.com/redhat-developer/devspaces/blob/devspaces-3-rhel-8/dependencies/che-plugin-registry/openvsx-sync.json[plugin-registry repository] to add an information about the extension:

Reviewer: ```suggestion
. Modify the plug-in registry's link:https://github.com/redhat-developer/devspaces/blob/devspaces-3-rhel-8/dependencies/che-plugin-registry/openvsx-sync.json[`openvsx-sync.json`] file to add an extension:
```

Reviewer: Note that this is a link to the downstream file... we probably need a new antora variable for this URL so we can pull either https://github.com/eclipse-che/che-plugin-registry/blob/main/openvsx-sync.json or https://github.com/redhat-developer/devspaces/blob/devspaces-3-rhel-8/dependencies/che-plugin-registry/openvsx-sync.json. 

Or should we link to the raw file instead?

https://raw.githubusercontent.com/eclipse-che/che-plugin-registry/main/openvsx-sync.json
vs.
https://raw.githubusercontent.com/redhat-developer/devspaces/devspaces-3-rhel-8/dependencies/che-plugin-registry/openvsx-sync.json

+
[source,json]
----
    {
        "id": "extension_id"
    }
----

> In this case the latest version of the extension from link:https://open-vsx.org/[Open VSX Registry] will be published. If need to specify the version, use the next structure:

Reviewer: ```suggestion
If no `version` is provided, the latest version of the extension from link:https://open-vsx.org/[Open VSX Registry] will be included in the registry. Alternatively, set a `version` to pin to a specific version:
```

+
[source,json]
----
    {
        "id": "extension_id",
        "version": "extension_version"
    }
----

> . Build the container image with plugin registry and push the image to a container registry:

Reviewer: ```suggestion
. To build the plug-in registry container image and publish it to a container registry, run the following commands:
```

+
[subs="+attributes,+quotes"]
----

> $ ./build.sh -o username -r quay.io -t next

Reviewer: ```suggestion
$ ./build.sh -o username -r quay.io -t custom
```

> $ {docker-cli} push username/che-theia-devspaces:next

Reviewer: ```suggestion
$ {docker-cli} push quay.io/username/che-plugin-registry:custom
```

Reviewer: for context see https://github.com/eclipse-che/che-plugin-registry/blob/main/build.sh#L128

----
. Customize an instance of `CheCluster` Custom Resource in the cluster to point the image and save the changes:
+
[source,yaml,subs="+quotes"]
----
spec:
  components:
    pluginRegistry:
      deployment:
        containers:

>           - image: quay.io/username/pluginregistry-rhel8:next

Reviewer: ```suggestion
          - image: quay.io/username/plugin-registry:custom
```

      openVSXURL: '' 
----

> . Check that `che-plugin-registry` pod restarted and it's running. Restart the workspace and check plugins in the Plugins view.

Reviewer: ```suggestion
. Check that `plugin-registry` pod restarted and is running. Restart the workspace and check the available extensions in the Extensions view.
```
